### PR TITLE
fix basic mobs retaliate target not filtering targets

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
basic mobs retaliate targeting would try to target players that they cant attack anymore,

## Why It's Good For The Game
fixes basic mobs retaliate targeting would try to target players that they cant attack anymore,

## Changelog
:cl:
fix: basic mobs retaliate targetting now selects targets they can attack
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
So we had a targetting datum but weren't using it?
I am mildly worried that this is changing the behaviour in an undesirable way, if we _weren't_ applying the targetting datum then we were ignoring faction checks, which feels like the expected behaviour in a lot of these cases.
You may need to go through and make sure that mobs we expect to retaliate against usually-allied creatures which attack them still do, or provide an alternate targetting datum to this behaviour on those, otherwise (for instance) mobs with the "neutral" faction will essentially never retaliate against anything.

**Comment by Ben10Omintrix** [comment]
> So we had a targetting datum but weren't using it? I am mildly worried that this is changing the behaviour in an undesirable way, if we _weren't_ applying the targetting datum then we were ignoring faction checks, which feels like the expected behaviour in a lot of these cases. You may need to go through and make sure that mobs we expect to retaliate against usually-allied creatures which attack them still do, or provide an alternate targetting datum to this behaviour on those, otherwise (for instance) mobs with the "neutral" faction will essentially never retaliate against anything.

same faction mobs were never able to retaliate against each other to begin with sadly. what would happen is the target would get assigned and they would chase after them endlessly but never actually attack them because the basic_melee behavior does a faction_check before calling melee_attack

**Comment by Jacquerel** [comment]
> > So we had a targetting datum but weren't using it? I am mildly worried that this is changing the behaviour in an undesirable way, if we _weren't_ applying the targetting datum then we were ignoring faction checks, which feels like the expected behaviour in a lot of these cases. You may need to go through and make sure that mobs we expect to retaliate against usually-allied creatures which attack them still do, or provide an alternate targetting datum to this behaviour on those, otherwise (for instance) mobs with the "neutral" faction will essentially never retaliate against anything.
> 
> same faction mobs and neutral mobs were never able to retaliate against each other to begin with sadly. what would happen is the target would get assigned and they would chase after them endlessly but never actually attack them because the basic_melee behavior does a faction_check before calling melee_attack.i will have to come up with a better solution

I see, unfortunate but makes sense.
Now that you remind me, I remember now that all of the "neutral retaliate" creatures just have a targetting datum which _never_ checks faction.

File: code/__DEFINES/ai/ai_blackboard.dm [hunk 1]
```diff
@@ -67,6 +67,8 @@
 
 ///List of mobs who have damaged us
 #define BB_BASIC_MOB_RETALIATE_LIST "BB_basic_mob_shitlist"
+///do we currently have a retaliate target?
+#define BB_BASIC_MOB_CURRENTLY_RETALIATTING "BB_basic_mob_currently_retaliating"
```
Comment (Jacquerel) [inline code/__DEFINES/ai/ai_blackboard.dm:71]: Maybe we could just call it `BASIC_MOB_IGNORE_FACTIONS` to be a bit more generic, you could have a different behaviour toggle this same blackboard key for some other reason (as long as you're careful not to have two behaviours compete with each other)

**Pull request merged**